src/components/table/_table.scss

```diff
@@ -125,6 +126,7 @@
     // Positioning of sticky headers
     .b-table-sticky-header > .table.b-table > thead > tr > th {
       // Header cells need to be sticky on top
+      border-top: 0;
```

Comment:
This assumes people are using `no-border-collapse` at the same time, which may not be the case.